bd-logger/src/async_log_buffer.rs

```diff
@@ -447,14 +447,24 @@ impl<R: LogReplay + Send + 'static> AsyncLogBuffer<R> {
   async fn process_log(&mut self, log: LogLine, block: bool) -> anyhow::Result<Vec<Log>> {
     // Prevent re-entrancy when we are evaluating the log metadata.
     let result = with_thread_local_logger_guard(|| {
-      self
-        .metadata_collector
-        .normalized_metadata_with_extra_fields(
-          log.fields,
-          log.matching_fields,
-          log.log_type,
-          &mut self.global_state_tracker,
-        )
+      if let Some(LogAttributesOverrides::PreviousRunSessionID(_id, _timestamp)) =
+        &log.attributes_overrides
+      {
+        // avoid normalizing metadata for logs from previous sessions, which may
+        // have had different global state
```

Comment:
I think this assumes that `PreviousRunSessionId` means that the caller handles global state but not sure if that's always true? This API is used by the AppExitLogger which I don't think handles the old global state, and we also probably need to think about whether we want the crash logs to be "naked" (aka no basic fields like app_version etc.) in the case where we failed to recover the old global state